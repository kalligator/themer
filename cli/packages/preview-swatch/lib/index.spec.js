const { render } = require('./index');
const { colors } = require('../../colors-default');

describe('themer color set swatch preview', () => {
  it('should use the color set name as the default name', (done) => {
    Promise.all(render(colors, { colors: 'colors-default' })).then((files) => {
      expect(
        files.every((file) =>
          /colors-default-(light|dark)-swatch\.svg/.test(file.name),
        ),
      ).toBe(true);
      done();
    });
  });
  it('should also accept custom names through options', (done) => {
    Promise.all(
      render(colors, {
        'themer-preview-swatch-name': 'test',
        'colors': 'colors-default',
      }),
    ).then((files) => {
      expect(
        files.every((file) => /test-(light|dark)-swatch\.svg/.test(file.name)),
      ).toBe(true);
      done();
    });
  });
  it('should render proper SVG files', (done) => {
    Promise.all(render(colors, { colors: 'colors-default' })).then((files) => {
      files.forEach((file) => {
        expect(file.contents.toString('utf8')).toMatchSnapshot();
      });
      done();
    });
  });
});
