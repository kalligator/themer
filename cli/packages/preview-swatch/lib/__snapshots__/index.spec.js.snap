// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer color set swatch preview should render proper SVG files 1`] = `
"
    <svg width=\\"480px\\" height=\\"160px\\" viewBox=\\"0 0 480 160\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\">
      <defs>
        <linearGradient x1=\\"0%\\" y1=\\"50%\\" x2=\\"100%\\" y2=\\"50%\\" id=\\"shade-scale\\">
          <stop stop-color=\\"#FFFCFF\\" offset=\\"0%\\"></stop><stop stop-color=\\"#E0DCE0\\" offset=\\"16.666666666666668%\\"></stop><stop stop-color=\\"#C1BCC2\\" offset=\\"33.333333333333336%\\"></stop><stop stop-color=\\"#A29DA3\\" offset=\\"50%\\"></stop><stop stop-color=\\"#847E85\\" offset=\\"66.66666666666667%\\"></stop><stop stop-color=\\"#656066\\" offset=\\"83.33333333333334%\\"></stop><stop stop-color=\\"#474247\\" offset=\\"100%\\"></stop>
        </linearGradient>
      </defs>
      <rect fill=\\"#282629\\" x=\\"0\\" y=\\"0\\" width=\\"480\\" height=\\"160\\"></rect>
      <circle fill=\\"#FF4050\\" cx=\\"51\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#F28144\\" cx=\\"105\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#FFD24A\\" cx=\\"159\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#A4CC35\\" cx=\\"213\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#26C99E\\" cx=\\"267\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#66BFFF\\" cx=\\"321\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#CC78FA\\" cx=\\"375\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#F553BF\\" cx=\\"429\\" cy=\\"49\\" r=\\"21\\"></circle>
      <rect fill=\\"url(#shade-scale)\\" x=\\"30\\" y=\\"86\\" width=\\"420\\" height=\\"42\\" rx=\\"3\\" ry=\\"3\\"></rect>
    </svg>
  "
`;

exports[`themer color set swatch preview should render proper SVG files 2`] = `
"
    <svg width=\\"480px\\" height=\\"160px\\" viewBox=\\"0 0 480 160\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\">
      <defs>
        <linearGradient x1=\\"0%\\" y1=\\"50%\\" x2=\\"100%\\" y2=\\"50%\\" id=\\"shade-scale\\">
          <stop stop-color=\\"#282629\\" offset=\\"0%\\"></stop><stop stop-color=\\"#474247\\" offset=\\"16.666666666666668%\\"></stop><stop stop-color=\\"#656066\\" offset=\\"33.333333333333336%\\"></stop><stop stop-color=\\"#847E85\\" offset=\\"50%\\"></stop><stop stop-color=\\"#A29DA3\\" offset=\\"66.66666666666667%\\"></stop><stop stop-color=\\"#C1BCC2\\" offset=\\"83.33333333333334%\\"></stop><stop stop-color=\\"#E0DCE0\\" offset=\\"100%\\"></stop>
        </linearGradient>
      </defs>
      <rect fill=\\"#FFFCFF\\" x=\\"0\\" y=\\"0\\" width=\\"480\\" height=\\"160\\"></rect>
      <circle fill=\\"#F03E4D\\" cx=\\"51\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#F37735\\" cx=\\"105\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#EEBA21\\" cx=\\"159\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#97BD2D\\" cx=\\"213\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#1FC598\\" cx=\\"267\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#53A6E1\\" cx=\\"321\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#BF65F0\\" cx=\\"375\\" cy=\\"49\\" r=\\"21\\"></circle><circle fill=\\"#EE4EB8\\" cx=\\"429\\" cy=\\"49\\" r=\\"21\\"></circle>
      <rect fill=\\"url(#shade-scale)\\" x=\\"30\\" y=\\"86\\" width=\\"420\\" height=\\"42\\" rx=\\"3\\" ry=\\"3\\"></rect>
    </svg>
  "
`;
